historian-server/src/main/java/com/hurence/webapiservice/timeseries/TimeSeriesExtracterImpl.java

```diff
@@ -21,17 +27,117 @@
 
     public TimeSeriesExtracterImpl(String metricName, long from, long to,
                                    SamplingConf samplingConf,
-                                   long totalNumberOfPoint) {
-        super(metricName, from, to, samplingConf, totalNumberOfPoint);
+                                   long totalNumberOfPoint,
+                                   List<AGG> aggregList) {
+        super(metricName, from, to, samplingConf, totalNumberOfPoint, aggregList);
         sampler = SamplerFactory.getPointSampler(this.samplingConf.getAlgo(), this.samplingConf.getBucketSize());
     }
 
     @Override
-    protected void samplePointsFromChunks(long from, long to, List<JsonObject> chunks) {
+    protected void samplePointsFromChunksAndCalculAggreg(long from, long to, List<JsonObject> chunks) {
         Stream<Point> extractedPoints = TimeSeriesExtracterUtil.extractPointsAsStream(from, to, chunks);
         Stream<Point> sortedPoints = extractedPoints
                 .sorted(Comparator.comparing(Point::getTimestamp));
-        List<Point> sampledPoints = sampler.sample(sortedPoints.collect(Collectors.toList()));
+        List<Point> points = sortedPoints.collect(Collectors.toList());
+        List<Point> sampledPoints = sampler.sample(points);
         this.sampledPoints.addAll(sampledPoints);
+        calculateAggreg(points);
+    }
+
+    protected void calculateAggreg(List<Point> points) {
+        if (!points.isEmpty())
+            aggregList.forEach(agg -> {
+                switch (agg) {
+                    case AVG:
+                        long numberOfPoint = points.size();
+                        long newNumberOfPoint = this.currentNumberOfPoints + numberOfPoint;
```

Comment:
this is good but there is no reason for having created a new variable when you have totalPointCounter that is exactly what you need. Just modify your code so that totalPointCounter is accurate once you are here.

Reply:
ok